starknet/src/bin.rs

```diff
@@ -44,28 +48,49 @@ async fn start(args: StartCommand) -> Result<()> {
     let mut node =
         StarkNetNode::<HttpProvider, SimpleRequestObserver, NoWriteMap>::builder(&args.rpc)?
             .with_request_observer(MetadataKeyRequestObserver::new("x-api-key".to_string()));
+    
+    match args.devnet {
+        true => {
+            let tempdir: TempDir;
+            tempdir = TempDir::new("apibara").unwrap();
+            node.with_datadir(tempdir.path().to_path_buf());
 
-    // give precedence to --data
-    if let Some(datadir) = args.data {
-        node.with_datadir(datadir);
-    } else if let Some(name) = args.name {
-        let datadir = default_data_dir()
-            .map(|p| p.join(name))
-            .expect("no datadir");
-        node.with_datadir(datadir);
-    }
+            // Setup cancellation for graceful shutdown
+            let cts = CancellationToken::new();
+            ctrlc::set_handler({
+                let cts = cts.clone();
+                move || {
+                    cts.cancel();
+                }
+            })?;
```

Comment:
I believe this doesn't need to belong inside the `match/if`. Can be safely used after the `match/if`.

My understanding is that you did it this way because you need to call `tempdir.close()` after running the node. I would try the following:

* remove `tempdir.close()` all together. The docs say that it will be called once the variable is dropped.
* if the variable is dropped too early, changed the code to something along the lines:

```rs
let mut devnet_dir = None;
if args.devnet {
  // created tempdir like you do now
  devnet_dir = Some(tempdir);
} else {
...
};

// start node
// after node.start(...) add this

if let Some(tempdir) = devnet_dir {
  tempdir.close()?;
}